### PR TITLE
feat: atlas-integration

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
## PR-Codex overview
This PR focuses on refactoring and enhancing user authentication and settings management in the application. It introduces new utility functions, updates toast notifications, and improves the handling of user settings and email verification.

### Detailed summary
- Deleted several unused files related to user authorization and settings.
- Added `isEmpty` utility function for string checks.
- Updated `debounceErrorToast` to use a new `errorToast` function.
- Refactored `AtlasProvider` to include new configuration.
- Modified `PreviousButton` to use `isEmpty` for route checks.
- Enhanced `NavBar` settings interface with an optional initial tab.
- Updated email regex for validation.
- Refactored `Search` component to utilize `isEmpty`.
- Introduced `InfoCard` component for displaying messages.
- Created a new `Settings` page with routing for email confirmation.
- Improved file upload handling in `handleFileUpload`.
- Integrated `EmailVerificationInfo` for email verification status.
- Updated `EnsureAuth` to streamline user sign-in process.
- Enhanced `FormContactDetails` to manage user settings more effectively.

> The following files were skipped due to too many changes: `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Release Notes

- **New Features**
	- Introduced `AtlasProvider` for enhanced context management.
	- Added `EmailVerificationInfo` component to display email verification status.
	- New `EmailConfirmation` component for handling email confirmation processes.
	- Added `InfoCard` component for displaying informational messages.
	- New utility function `timeLeftUntil` for calculating time until a specified date.
	- Added `isEmpty` utility function for string validation.

- **Improvements**
	- Enhanced email validation logic with a more comprehensive regex.
	- Streamlined authentication processes in the `EnsureAuth` component.
	- Improved user feedback mechanisms with toast notifications across several components.
	- Updated the `NextButton` and `PreviousButton` components to utilize the `isEmpty` utility for better validation.

- **Bug Fixes**
	- Addressed various issues related to user settings and email management functionalities.

- **Chores**
	- Cleaned up unused dependencies and updated existing packages in `package.json`.

### CODE REVIEW

File: web/src/pages/NewTransaction/NavigationButtons/NextButton.tsx [hunk 1]
```diff
@@ -68,49 +71,69 @@ const NextButton: React.FC<INextButton> = ({ nextRoute }) => {
...
-          setTransactionUri(transactionUri);
+    if (location.pathname.includes("/new-transaction/deliverable") && escrowType === "general") {
+      const transactionUri = await handleFileUpload(
+        uploadFile,
+        escrowTitle,
+        deliverableText,
+        setIsFileUploading,
+        setExtraDescriptionUri,
+        deliverableFile
+      );
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/NewTransaction/NavigationButtons/NextButton.tsx:85]: handled within function

File: web/src/layout/Header/navbar/index.tsx [hunk 2]
```diff
@@ -53,6 +53,7 @@ const DisconnectWalletButtonContainer = styled.div`
 
 export interface ISettings {
   toggleIsSettingsOpen: () => void;
+  initialTab?: number;
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/index.tsx:56]: it's optional

File: web/src/pages/Settings/EmailConfirmation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  expired: {
+    headerMsg: "Verification link expired...",
+    subtitleMsg:
+      "Oops, the email verification link has expired. No worries! Go to settings and click on Resend it to receive another verification email.",
+    buttonMsg: "Open Settings",
+    buttonTo: "/#notifications",
+    Icon: WarningIcon,
+    color: "warning",
+  },
+};
```
Comment (alcercu) [inline web/src/pages/Settings/EmailConfirmation/index.tsx:128]: Maybe move this into its own file on this same folder? 🤔

**Review by kemuru — APPROVED** [review]
lgtm

**Pull request merged**